### PR TITLE
style: update Menu component

### LINKED ISSUE #223: Update Select Menu molecule to conform to the styleguide
The Select Menu molecule should look like the one in the [styleguide](https://projects.invisionapp.com/share/NZKXOQUV4P3#/screens/302926985).

Right now we don't have any Select Menu component. We could start from [Subject Area Dropdown](https://github.com/elifesciences/elife-xpub/blob/develop/app/components/submission/ManuscriptMetadata/SubjectAreaDropdown.js) from our own repository or the [Menu](https://gitlab.coko.foundation/pubsweet/pubsweet/blob/master/packages/ui/src/molecules/Menu.js) from pubsweet.

### PR DESCRIPTION
**Pull request by tamlyn**
#### Background

- Fixes #223 
- Ready for review ~~but do not merge until [pubsweet!304](https://gitlab.coko.foundation/pubsweet/pubsweet/merge_requests/304) is merged and published~~

#### How has this been tested?

Visually ~~when linked to PubSweet master.~~

### CODE REVIEW

File: client/elife-theme/src/elements/Menu.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    &:hover {
+      border-color: ${th('colorBorder')};
+    }
+    ${props =>
+      props.open &&
+      `border-bottom-left-radius: 0; border-bottom-right-radius: 0;`};
+  `,
+  Option: css`
+    height: ${th('space.5')};
+    line-height: ${th('space.5')};
```
Comment (jsms90) [inline client/elife-theme/src/elements/Menu.js]: Has there been a decision to use `space` instead of the `fontLineHeightXYZ` variables in our theme?

File: client/elife-theme/src/elements/Menu.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    border-radius: 0 0 ${th('borderRadius')} ${th('borderRadius')};
+  `,
+  Value: css`
+    &:hover {
+      color: ${th('colorText')};
+    }
+  `,
+  Placeholder: css`
+    font-style: normal;
+    color: ${th('colorText')};
```
Comment (jsms90) [inline client/elife-theme/src/elements/Menu.js]: `colorTextPlaceholder`?

File: client/elife-theme/src/elements/Menu.js (hunk 1 above)
Comment (tamlyn) [reply client/elife-theme/src/elements/Menu.js]: In this instance line-height is being used for layout (to centre the text in its box) rather than typography.

**Comment by tamlyn** [comment]
All inputs should be 100% width and the parent should control the layout. Also, this change has already been made but I guess you can't see that as the CI is playing up again. Ugh.

**Comment by hdrury1** [comment]
@tamlyn Can you post a screenshot for reference of this change?

**Comment by tamlyn** [comment]
Review deployment now working http://review-menu-overr-m8sm7l.gateway.xpub.semioticsquares.com/submit/metadata

**Comment by jsms90** [comment]
Might be nice to have the scroll bar removed is/when the list isn't long enough to require it
image

Note: this only happens on Firefox

**Comment by jsms90** [comment]
Also, @tamlyn the dropdown icon is still larger than the invision designs.

But we could merge it as is & raise tech debt issues, so @hdrury1 can show to the design team? :woman_shrugging:

**Comment by jsms90** [comment]
Ah, we've just realised that the down arrow is only large on firefox & actually the down arrow looks tiny in Chrome (on my computer only).

Strange things. But seems to be largely browser specific & I have a feeling that pubsweet + firefox doesn't actuallly play that well anyway.

Here's the elifesciences.org homepage on chrome:
image

and here it is on firefox:
image

:woman_shrugging: 

Happy to merge now & raise issues about browser support, wherever necessary

**Comment by tamlyn** [comment]
I'd like to specify that the component and site in general looks fine in Firefox on Mac. It appears to just be Firefox on Linux that is affected by these issues.

File: client/elife-theme/src/elements/ui/Menu.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import { css } from 'styled-components'
+import { th } from '@pubsweet/ui-toolkit'
+
+export default {
+  Arrow: css`
+    color: ${th('colorTextSecondary')};
+    transition: none;
+    font-size: ${th('gridUnit')};
```
Comment (dryhten) [inline client/elife-theme/src/elements/ui/Menu.js]: space?

File: client/elife-theme/src/index.js [hunk 4]
```diff
@@ -6,6 +6,7 @@ const gridUnit = 6
 
 export default {
   /* Colors */
+  colorBackground: '#fff',
```
Comment (dryhten) [inline client/elife-theme/src/index.js]: Why did you add this? You don't use it in here (`background-color` is set to `colorPrimary`). We removed it because it was not on `VARIABLES.md` but it is now on the wiki. We don't have it in invision.

Comment (tamlyn) [reply client/elife-theme/src/index.js]: It's used in PubSweet components so if we don't have it those components don't render properly. Yet another reason why `VARIABLES.md` should be deleted.

Comment (dryhten) [reply client/elife-theme/src/index.js]: I just checked and yes, it's being currently used. Makes sense to add it.

File: client/elife-theme/src/elements/ui/Menu.js (hunk 3 above)
Comment (jsms90) [reply client/elife-theme/src/elements/ui/Menu.js]: Having 2 variables in our theme which evaluate to the same thing (`gridUnit` and `space.1`) seems less than ideal to me. Also for `calc(...something based on gridUnit...)` vs `space.4`/`space.3`/whatever.

Seems like we need a team-wide convention. Perhaps `gridUnit` is fine for now @dryhten?

**Comment by tamlyn** [comment]
@hdrury1 could you please re-review this? It's blocked until you do because you requested changes.
